src/components/FilterTags/FilterTags.js

```diff
@@ -87,20 +88,95 @@ const FilterTags = () => {
     };
   }, []);
 
-  const handleFilterClick = () => {
-    console.log("Filter clicked");
+  /**
+  Apply a given filter
+  @param {object} e The event object
+  @param {string} filter The name of the filter
+   @param {string} filterBy The name of the filter group
+*/
+  const addActiveFilter = (e, filter, filterBy) => {
+    setActiveFilters(filters => {
+      const updatedFilters = { ...filters };
+      updatedFilters[filterBy] = updatedFilters[filterBy] || [];
+      if (!updatedFilters[filterBy].includes(filter)) {
+        updatedFilters[filterBy].push(filter);
+      }
+      return updatedFilters;
+    });
+    e.currentTarget.classList.add("is-selected");
   };
 
+  /**
+  Remove a given filter
+  @param {object} e The event object
+  @param {string} filter The name of the filter
+   @param {string} filterBy The name of the filter group
+*/
+  const removeActiveFilter = (e, filter, filterBy) => {
+    e.stopPropagation();
+    setActiveFilters(filters => {
+      const updatedFilters = {
+        ...filters
+      };
+      if (updatedFilters[filterBy].includes(filter)) {
+        const index = updatedFilters[filterBy].indexOf(filter);
+        updatedFilters[filterBy].splice(index, 1);
+      }
+      return updatedFilters;
+    });
+
+    // Remove is-selected state from relevant filter from filter panel
+    const panelFilterItems = document.querySelectorAll(
+      ".p-filter-panel__button"
+    );
+    panelFilterItems.forEach(panelFilterItem => {
+      if (panelFilterItem.innerHTML === filter) {
+        panelFilterItem.classList.remove("is-selected");
+      }
+    });
+  };
+
+  let activeFiltersCount = 0;
```

Comment:
If you simplify the storage of the filters this can be simplified to just `activeFilters.size`